src/scripts/helpers/formControllerHelpers.ts

```diff
@@ -0,0 +1,171 @@
+import type { Request, Response, NextFunction } from 'express';
+import 'csrf-sync'; // Import to ensure CSRF types are loaded
+import { apiService } from '#src/services/apiService.js';
+import { safeString, validateForm, capitaliseFirst, extractCurrentFields } from '#src/scripts/helpers/index.js';
+import { type Result, validationResult } from 'express-validator';
+import { formatValidationError, type ValidationErrorData } from '#src/scripts/helpers/ValidationErrorHelpers.js';
+import type {
+  RenderData,
+  GetFormOptions,
+  PostFormOptions,
+  ValidationResult
+} from '#types/form-controller-types.js';
+
+const BAD_REQUEST = 400;
+const MINIMUM_FIELDS_LENGTH = 0;
+
+/**
+ * Generic function to handle GET requests for edit forms
+ * @param {Request} req - Express request object
+ * @param {Response} res - Express response object
+ * @param {NextFunction} next - Express next middleware function
+ * @param {GetFormOptions} options - Configuration options for the form handler
+ * @returns {Promise<void>}
+ */
+export async function handleGetEditForm(
+  req: Request,
+  res: Response,
+  next: NextFunction,
+  options: GetFormOptions
+): Promise<void> {
+  const caseReference = safeString(req.params.caseReference);
+  try {
+    const response = await apiService.getClientDetails(req.axiosMiddleware, caseReference);
+
+    let extractedData: Record<string, unknown> = {};
+    if (options.fieldConfigs !== undefined) {
+      // Use field configurations for data extraction
+      extractedData = extractCurrentFields(response.data, options.fieldConfigs);
+    } else if (options.dataExtractor !== undefined) {
+      // Use custom data extractor function
+      extractedData = options.dataExtractor(response.data);
+    }
+
+    const templateData = {
+      caseReference,
+      csrfToken: typeof req.csrfToken === 'function' ? req.csrfToken() : undefined,
+      ...extractedData
+    };
+    res.render(options.templatePath, templateData);
+  } catch (error) {
+    next(error);
+  }
+}
+
+/**
+ * Generic function to handle POST requests for edit forms
+ * @param {Request} req - Express request object
+ * @param {Response} res - Express response object
+ * @param {NextFunction} next - Express next middleware function
+ * @param {PostFormOptions} options - Configuration options for the form handler
+ * @returns {Promise<void>}
+ */
+export async function handlePostEditForm(
+  req: Request,
+  res: Response,
+  next: NextFunction,
+  options: PostFormOptions
+): Promise<void> {
+  const caseReference = safeString(req.params.caseReference);
+  const { templatePath, fields, apiUpdateData, useCustomValidation = false } = options;
+
+  let inputErrors: Record<string, string> = {};
+  let errorSummaryList: Array<{ text: string; href: string }> = [];
+  let formIsInvalid = false;
+
+  if (useCustomValidation) {
```

Comment:
Question - Will we need to refactor this out once we are using `express-validation` for everything? Maybe worth adding that refactor to EL-2572?

Reply:
I have changed `useCustomValidation` to `useDefaultValidator` and made it `true` by default (so `express-validator` is now the standard). Flipped the logic so most controllers don't need to define it, only exceptional cases like DOB implementation that require complex validation handling need to set `useDefaultValidator: false`

Reply:
Yeah, I think that way round makes more sense. 👍🏽